File: aat-lib/src/main/java/ch/bailu/aat_lib/preferences/StorageInterface.java
package ch.bailu.aat_lib.preferences;

public interface StorageInterface {
    String readString(String key);
    void writeString(String key, String value);
    int readInteger(String key);
    void writeInteger(String key, int v);
    void writeIntegerForce(String key, int v);

    long readLong(String key);
    void writeLong(String key, long v);

    void register(OnPreferencesChanged listener);
    void unregister(OnPreferencesChanged l);

    boolean isDefaultString(String s);
    String getDefaultString();
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/preferences/StorageInterface.kt
package ch.bailu.aat_lib.preferences

interface StorageInterface {
    fun readString(key: String): String
    fun writeString(key: String, value: String)
    fun readInteger(key: String): Int
    fun writeInteger(key: String, value: Int)
    fun writeIntegerForce(key: String, value: Int)
    fun readLong(key: String): Long
    fun writeLong(key: String, value: Long)
    fun register(onPreferencesChanged: OnPreferencesChanged)
    fun unregister(onPreferencesChanged: OnPreferencesChanged)
    fun isDefaultString(s: String): Boolean
    fun getDefaultString(): String
}
